src/ScriptExecutor.php

```diff
@@ -116,20 +117,8 @@ private function expandPhpScript(string $cmd, array $scriptArguments): string
 
         $arguments = $phpFinder->findArguments();
 
-        if ($env = (string) getenv('COMPOSER_ORIGINAL_INIS')) {
-            $paths = explode(\PATH_SEPARATOR, $env);
-            $ini = array_shift($paths);
-        } else {
-            $ini = php_ini_loaded_file();
-        }
-
-        if ($ini) {
-            $arguments[] = '--php-ini='.$ini;
-        }
-
-        if ($memoryLimit = (string) getenv('COMPOSER_MEMORY_LIMIT')) {
```

Comment:
I think we still needs this if block, the env value is added as argument to the processes which is not done via PhpConfig

Reply:
but the memory limit has been configured and dumped into the tmp ini file, so with persistent settings those changes done to the restarted PHP process should be preserved (I think)

Reply:
Nope, XdebugHandler dumped the ini value not the runtime value.

It is not possible to set the PHP memory limit via envs (beside when configured in php.ini). So custom code like this is required.

This code makes it also possible to have another memory limit for a subprocess like Composer then the main processes.
But you can debate if it is correct that this env will be passed to a subprocesses. If you give Composer a limit of 1Gb, you will give the main processes and each subprocess 1Gb, so it is possible that the whole run takes 2Gb.

Reply:
I do not understand what you mean. If you executed composer with a specific memory limit, Composer will have restarted with persistent settings where the current memory limit (and the one that XdebugHandler will use) is that one.

But I admit I think it deserves to be properly checked rather than merged blindly. It needs a fix, but we should make sure this change is the right fix.